sdk/core/azure-core-amqp/src/amqp/private/message_sender_impl.hpp

```diff
@@ -5,34 +5,41 @@
 
 #include "azure/core/amqp/message_sender.hpp"
 #include "claims_based_security_impl.hpp"
+#include "link_impl.hpp"
 #include <azure_uamqp_c/message_sender.h>
 #include <tuple>
 
+template <> struct Azure::Core::_internal::UniqueHandleHelper<MESSAGE_SENDER_INSTANCE_TAG>
```

Comment:
It is better to have implementation for this in an unnamed namespace - I've done so in Identity. If it is in unnamed namespace, and two different libraries happen to instantiate the same template with the same parameters, there won't be a linker conflict.
If you agree, please do this for all instances - I think I saw more template instantiations.
If you don't agree, let's talk later - I think I can make an example showing how the names can collide, after which there is really no reasons to instantiate the template in global namespace, unnamed namespace is better in all aspects.

Reply:
The compiler won't let me declare this in an anonymous namespace.

It appears that identity has cloned the implementation of Azure::Core:;-internal::UniqueHandleHelper, AND it's located in a .cpp file, which won't work in this case (since it has to be in a header).

The good news is that unlike Win32, the uAMQP APIs all have unique C identities. So if someone else instantiates the template with the same parameters, it is an authoring error (unless you are contemplating a world where there are multiple C++ package which are built on top of uAMQP).

If we do end up with multiple C++ SDK packages which directly use uAMQP, I'll figure out a solution when that happens. Otherwise I'd rather use our standardized mechanism for handling unique types.